.github/workflows/ci.yml

```diff
@@ -0,0 +1,84 @@
+name: CI
+on: [pull_request, push]
+jobs:
+  test:
+    name: Test
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [macos-latest, ubuntu-latest, windows-latest]
+        toolchain: [stable, nightly]
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup Rust
+        run: |
+          rustup toolchain add ${{ matrix.toolchain }} --component llvm-tools-preview
+          rustup override set ${{ matrix.toolchain }}
+      - name: Configure cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
```

Comment:
Why caching `target/`? So we can do incremental compilation?

Reply:
Pretty much yes. It allows to run the test much sooner as we don't have to re-compile all the dependencies over and over again.

Reply:
It could be problematic for coverage calculation. We need to ensure we save coverage artifacts outside that directory